src/torchcodec/decoders/_core/CPUOnlyDevice.cpp

```diff
@@ -13,7 +14,19 @@ void throwUnsupportedDeviceError(const torch::Device& device) {
   throw std::runtime_error("Unsupported device: " + device.str());
 }
 
-void initializeDeviceContext(const torch::Device& device) {
+VideoDecoder::DecodedOutput convertAVFrameToDecodedOutputOnDevice(
+    const torch::Device& device,
+    const VideoDecoder::VideoStreamDecoderOptions& options,
+    AVCodecContext* codecContext,
+    VideoDecoder::RawDecodedOutput& rawOutput) {
+  throwUnsupportedDeviceError(device);
```

Comment:
Should this function and the function below always throw? If yes, then we should just do something like `TORCH_CHECK(false, "Unsupported device.");`. In order avoid the need for a return value, mark the function as `[[noreturn]]`: https://en.cppreference.com/w/cpp/language/attributes/noreturn. We should rely on a TORCH macro to do the throwing for us rather than doing the throw ourselves, and we should make it obviously one that will always fail its check.

Reply:
Good suggestion. Done

Reply:
I think we maybe should also annotate `convertAVFrameToDecodedOutputOnDevice()` and `initializeDeviceContext()` with `[[noreturn]]`. Let's also avoid two `TORCH_CHECK` calls. Whatever message we want to put on stderr, we can do it in one check.

Reply:
The two checks are there because one is a programming/logic error on our part -- we should never pass in a CPU device for device functions.

The other is the check for passing in a non-compiled device.